apps/api_web/lib/api_web/controllers/route_pattern_controller.ex

```diff
@@ -0,0 +1,186 @@
+defmodule ApiWeb.RoutePatternController do
+  @moduledoc """
+  Controller for Routes. Filterable by:
+
+  * id
+  * route_id
+  """
+  use ApiWeb.Web, :api_controller
+  alias State.RoutePattern
+
+  @filters ~w(id route)
+  @pagination_opts [:offset, :limit, :order_by]
+  @description """
+  Route patterns are used to describe the subsets of a route, representing different possible patterns of where trips may serve. For example, a bus route may have multiple branches, and each branch may be modeled as a separate route pattern per direction. Hierarchically, the route pattern level may be considered to be larger than the trip level and smaller than the route level.
+
+  For most MBTA modes, a route pattern will typically represent a unique set of stops that may be served on a route-trip combination. Seasonal schedule changes may result in trips within a route pattern having different routings. In simple changes, such a single bus stop removed or added between one schedule rating and the next (for example, between the Summer and Fall schedules), trips will be maintained on the same route_pattern_id. If the changes are significant, a new route_pattern_id will be introduced.
+
+  For Commuter Rail, route patterns represent complete branches at this time, with most including trips on multiple sets of stops. For example, CR-Providence will contain two route patterns per direction, one for the Wickford Junction branch and the other for the Stoughton branch.
```

Comment:
I don't understand this part. Maybe something like "each branch will have a separate route pattern"? There can be more patterns than branches:
- Kingston has 6 patterns
- Franklin has 3 patterns